communication/nats/receiver.go

```diff
@@ -66,12 +70,28 @@ func (receiver *receiverNATS) Receive(consumer communication.MessageConsumer) er
 		}
 	}
 
-	_, err := receiver.connection.Subscribe(messageTopic, messageHandler)
+	receiver.mu.Lock()
+	defer receiver.mu.Unlock()
+
+	subs, err := receiver.connection.Subscribe(messageTopic, messageHandler)
```

Comment:
might be a nitpick, but it returns a sub, not subs(the s on the end indicates a plural, in my head at least)

Reply:
correct, fixed it.